### PR TITLE
Fix mrpt-reactivenav2d compilation errors

### PR DESCRIPTION
**Pull request by bergercookie**
PR fixes https://github.com/mrpt-ros-pkg/mrpt_navigation/issues/35

### CODE REVIEW

File: mrpt_reactivenav2d/src/mrpt_reactivenav2d_node.cpp [hunk 1]
```diff
@@ -185,8 +192,10 @@ class ReactiveNav2DNode
 #if MRPT_VERSION>=0x150
 		bool stop()
 		{
-			std::vector<double> cmds(2,0.0);
-			return changeSpeeds(cmds);
+			mrpt::kinematics::CVehicleVelCmd_DiffDriven vel_cmd;
+			vel_cmd.lin_vel = 2;
```
Comment (jlblancoc) [inline mrpt_reactivenav2d/src/mrpt_reactivenav2d_node.cpp]: This should be: 
vel_cmd.lin_vel = 0;

the "2" before in `cmds(2,0.0);` was due to the length of the vector, and all of the elements set to "0.0"....
Not clear at all, hence the refactoring to the new structure in which each velocity component is referenced by name ;-)

File: mrpt_reactivenav2d/src/mrpt_reactivenav2d_node.cpp [hunk 2]
```diff
@@ -122,7 +125,7 @@ class ReactiveNav2DNode
 		 * \return false on any error.
 		 */
 #if MRPT_VERSION>=0x150
-		bool getCurrentPoseAndSpeeds(mrpt::math::TPose2D &curPose, mrpt::math::TTwist2D &curVel)
+		bool getCurrentPoseAndSpeeds(mrpt::math::TPose2D &curPose, mrpt::math::TTwist2D &curVel, mrpt::system::TTimeStamp &timestamp)
 		{
```
Comment (jlblancoc) [inline mrpt_reactivenav2d/src/mrpt_reactivenav2d_node.cpp]: output param `timestamp` should be set to either: 
1)  `mrpt::system::now()` (and add a `MRPT_TODO("get real timestamp")`, or 
2) to the timestamp of where TF says that was the robot pose.

File: mrpt_reactivenav2d/src/mrpt_reactivenav2d_node.cpp [hunk 3]
```diff
@@ -207,7 +216,7 @@ class ReactiveNav2DNode
 
 		/** Return the current set of obstacle points.
 		  * \return false on any error. */
-		virtual bool senseObstacles( CSimplePointsMap  &obstacles )
+		virtual bool senseObstacles( CSimplePointsMap  &obstacles, mrpt::system::TTimeStamp &timestamp)
 		{
```
Comment (jlblancoc) [inline mrpt_reactivenav2d/src/mrpt_reactivenav2d_node.cpp]: ídem as in getCurrentPoseAndSpeeds() about the timestamp.

**Comment by jlblancoc** [comment]
I intended to check this tonight, but you were faster, thanks! 👍 

Please, note the 3 comments above. If you think you can yourself fix them, go on and add more commits before accepting the PR, otherwise I could do it later... let me know!

**Comment by bergercookie** [comment]
Thanks for the remarks.
I think I got it..
I 'll just have to get back home and get to the computer.. :)

**Comment by jlblancoc** [comment]
Thanks for all the commits! :-) 
Let me know when you test it to be OK.

**Comment by bergercookie** [comment]
>Let me know when you test it to be OK.

I don't know exactly what you mean by "test" but I think its OK, at least on Indigo with MRPT 1.5.0 :)

**Comment by jlblancoc** [comment]
👍

**Pull request merged**